### ISSUE TITLE
diskstats: Naming of disks.

### ISSUE THREAD

**Issue by simonnagl** [firehol/netdata#435]
### At the moment we have the following naming conventions for disks in the plugin `proc_diskstats`:
- If we find a mount point, the name is the absolute path of the mount point. There is no handling of multiple mount points.
- If not we take column three of `/proc/diskstats`.
### This is a list of which names exist for disks.
- **Major:Minor**: Unique identifier of the disk.
- **Device Name:** Unique name for the disk
  - Third column of `proc/diskstats` (For example `sda`, `dm-0`)
- **Mountpoint(s):** Not available for all disks. One disk may have multiple mount points.
  - Fifth column of `proc/self/mountinfo`
- **Device Mapper name:** Available for virtual disks. 
  - `/sys/block/<device name>/dm/name`
  - `/sys/dev/block/<maj:min>/dm/name`
- **Device Model:** Available for physical disks. (For example SanDisk SSD U100)
  - `/sys/block/<device name>/device/model`
  - `/sys/dev/block/<maj:min>/device/model`
### My proposal for the naming convention:
- Physical disks and partitions: `<device name> (<mount points>)`
- Virtual disks: `<Device mapper name> (<mount points>)`

**Comment by ktsaou** [comment]
Seems nice.

Keep in mind some names break JSON and need to be escaped/mapped/converted.

netdata supports the following:

1. charts ids
2. chart names
3. family names (the submenus on the dashboard)

I suggest chart ids and chart names to always be device names. Only family names should be affected.

**Comment by simonnagl** [comment]
> I suggest chart ids and chart names to always be device names. Only family names should be affected.

This was my intention.

**Comment by robbat2** [comment]
Device names might not be consistent between boots, or even during the duration of a given instance running.

In my case, I want to override the alarm for a specific mountpoint. It is on LVM, and does not have a consistent device number between boots. This boot it was `dm-25`, and last boot it was `dm-23`, so my alarm override stopped matching.

**Comment by ktsaou** [comment]
I am not sure there anything to do for this currently.

netdata now has 2 disk plugins:

1. disk performance metrics
2. disk space monitoring

When the 2 match (same disk), netdata uses the mount point as the family of both (so both have the same name at the dashboard.

I am closing this. If you believe there is something missing, please post and I'll reopen it.

**Comment by zeleznikm** [comment]
After every reboot my disks have different _dm-number_.
plugin diskstats are enabled for my disk and backend server is also enabled.

There would be great option to assign unique disk name to disk _dm-number_ to collect data on backend server for the same disk

like:
[plugin:proc:/proc/diskstats:dm-7]
         enable = yes
         enable performance metrics = yes
         bandwidth = auto
         operations = auto
         merged operations = auto
         i/o time = auto
         queued operations = auto
         utilization percentage = auto
         backlog = auto
         **backlog name = disk1**

**Comment by ktsaou** [comment]
hm... this won't help, since on every update the `dm-number` will be different, so the configuration you have above might not be applied.

If the disk is mounted, netdata also assigns a family to it taken from the mount point.

```json
		"disk.md127": 		{
			"id": "disk.md127",
			"name": "disk.md127",
			"type": "disk",
			"family": "/data",
			"context": "disk.io",
			"title": "Disk I/O Bandwidth (disk.md127)",
			"priority": 2000,
			"enabled": true,
			"units": "kilobytes/s",
			"data_url": "/api/v1/data?chart=disk.md127",
			"chart_type": "area",
			"duration": 29596,
			"first_entry": 1492469068,
			"last_entry": 1492498664,
			"update_every": 1,
			"dimensions": {
				"reads": { "name": "reads" },
				"writes": { "name": "writes" }
			},
			"green": null,
			"red": null
		},
```

I see in graphite, opentsdb and prometheus this information is not sent to the backend server. I am not sure how/where to add it... In json backends, it is sent.

So, netdata already has a name for it, but I am not sure how this information should be send to the backend servers.

**Comment by zeleznikm** [comment]
![netdata1](https://cloud.githubusercontent.com/assets/22667061/25120619/2b13e852-241f-11e7-8248-73931238ba1a.jpg)

http://xx.x.xxx.xxx:8086/query?q=SHOW+MEASUREMENTS&db=netdata

> .
> .
> .
> 
> netdata.disk.dm-7.reads
> netdata.disk.dm-7.writes
> .
> .
> 

Maybe to send data to backend with Family name
like:
netdata.disk.home.db2inst1.data1.DB2T24L.reads
netdata.disk.home.db2inst1.data1.DB2T24L.writes

**Comment by ktsaou** [comment]
What you suggest is to replace the device name with the mount point. Unfortunately, the way netdata is now, this will be based on heuristics. I don't have a proper way of doing it, without hardcoding rules for disks, network interfaces, etc.

To do it properly, it should be implemented after #807
